src/Collections/Artemis.Plugins.Audio/DataModelExpansion/PlaybackVolumeDataModelExpansion.cs

```diff
@@ -0,0 +1,217 @@
+            {
+                // Absolute master peak volume 
+                float peakVolumeNormalized = (float) _playbackDevice?.AudioMeterInformation.MasterPeakValue;
+
+                // Sound detected. Reset timespan
+                if (peakVolumeNormalized > 0)
+                {
+                    DataModel.TimeSinceLastSound = TimeSpan.Zero;
+                }
+
+                // Don't update datamodel if not neeeded
+                if (_lastMasterPeakVolumeNormalized == peakVolumeNormalized)
+                    return;
+
+                DataModel.PeakVolumeNormalized = _lastMasterPeakVolumeNormalized = peakVolumeNormalized;
+                DataModel.PeakVolume = peakVolumeNormalized * 100f;
+
+                // Master peak volume relative to master volume
+                DataModel.PeakVolumeRelativeNormalized = peakVolumeNormalized * DataModel.VolumeNormalized;
+                DataModel.PeakVolumeRelative = peakVolumeNormalized * 100f * DataModel.VolumeNormalized;
+
+                // Update Channels Peak
+                AudioMeterInformationChannels channelsVolumeNormalized = _playbackDevice?.AudioMeterInformation.PeakValues;
+
+                //One more check because Playback device can be null any time (device for example). If this is the case, just keep the actual values and update in the next update.
+                if (channelsVolumeNormalized == null)
+                    return;
+
+                for (int i = 0; i < DataModel.Channels.DynamicChildren.Count; i++)
+                {
+                    DynamicChild<ChannelDataModel> channelDataModel = DataModel.Channels.GetDynamicChild<ChannelDataModel>(i.ToString());
+                    channelDataModel.Value.PeakVolumeNormalized = channelsVolumeNormalized[i];
+                    channelDataModel.Value.PeakVolume = channelsVolumeNormalized[i] * 100f;
+                }
+            }
+        }
+
+        private void UpdateVolumeDataModel()
+        {
+            DataModel.VolumeChanged.Trigger();
+            DataModel.VolumeNormalized = (_audioEndpointVolume.MasterVolumeLevelScalar);
+            DataModel.Volume = DataModel.VolumeNormalized * 100f;
+            DataModel.ChannelCount = _audioEndpointVolume.Channels.Count;
+            DataModel.DeviceState = _playbackDevice.State;
+            DataModel.Muted = _audioEndpointVolume.Mute;
+
+            lock (_audioEventLock)
+            {
+                for (int i = 0; i < _audioEndpointVolume.Channels.Count; i++)
+                {
+                    DynamicChild<ChannelDataModel> channelDataModel = DataModel.Channels.GetDynamicChild<ChannelDataModel>(i.ToString());
+                    float volumeNormalized = _audioEndpointVolume.Channels[i].VolumeLevelScalar;
+                    channelDataModel.Value.VolumeNormalized = volumeNormalized;
+                    channelDataModel.Value.Volume = volumeNormalized * 100f;
+                }
```

Comment:
It would be better to keep dynamic children in a list instead of this. `AddDynamicChild` returns the newly created dynamic child, please iterate that instead of this for loop.

It's not much but it creates some extra overhead